crates/storage/src/address.rs

```diff
@@ -35,9 +34,11 @@ use crate::env::random_bytes;
 /// system operation. Abstracting the true type away provides a level of
 /// insulation that is useful for any future changes.
 ///
-#[derive(Copy, Clone, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
+#[derive(BorshSerialize, BorshDeserialize, Copy, Clone, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
 #[serde(crate = "calimero_sdk::serde")]
-pub struct Id(Uuid);
+pub struct Id {
+    pub bytes: [u8; 32],
```

Comment:
keep it private, or at least impl `Deref` so `*id` works.

but we shouldn't allow direct access to the internal field

Reply:
in my [later comment](https://github.com/calimero-network/core/pull/907#discussion_r1824465398), `From<Id> for [u8; 32]` is fine, no need for `Deref`